Source/Scene/ImplicitAvailabilityBitstream.js

```diff
@@ -0,0 +1,76 @@
+import Check from "../Core/Check.js";
+import defined from "../Core/defined.js";
+import DeveloperError from "../Core/DeveloperError.js";
+import RuntimeError from "../Core/RuntimeError.js";
+
+/**
+ * An availability bitstream for use in an {@link ImplicitSubtree}. This handles
+ * both Uint8Array bitstreams and constant values.
+ *
+ * @alias ImplicitAvailabilityBitstream
+ * @constructor
+ *
+ * @param {Object} options An object with the following properties:
+ * @param {Number} options.lengthBits The length of the bitstream in bits
+ * @param {Boolean} [options.constant] A single boolean value indicating the value of all the bits in the bitstream if they are all the same
+ * @param {Uint8Array} [options.bitstream] An array of bytes storing the bitstream in binary
+ * @param {Number} [options.availableCount] A number indicating how many 1 bits are found in the bitstream
+ * @private
+ */
+export default function ImplicitAvailabilityBitstream(options) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.number("options.lengthBits", options.lengthBits);
+  //>>includeEnd('debug');
+
+  this._availableCount = options.availableCount;
+  this._lengthBits = options.lengthBits;
+  this._constant = undefined;
+  this._bitstream = undefined;
+
+  if (defined(options.constant)) {
+    this._constant = options.constant;
+  } else {
+    var expectedLength = Math.ceil(this._lengthBits / 8);
+    if (options.bitstream.length !== expectedLength) {
+      throw new RuntimeError(
+        "options.bitstream.length must have ceil(options.lengthBits / 8) bytes."
```

Comment:
Since this is a runtime error that the user would see the message should describe the error in a more generic way explaining how the data is wrong rather than mentioning implementation details like `options.bitstream.length`